### PR TITLE
security: restrict syscall arguments

### LINKED ISSUE #656: Add seccomp restriction on syscall argument values
Currently we use seccomp to create an allow list of syscalls, but then all arguments to those syscalls are permitted.

We should add functionality to filter syscall arguments as well (eg, the write syscall limited only to descriptor 3, stderr), and implement that across all allowed syscalls.

### PR DESCRIPTION
**Pull request by mmcgee-jump**
fd_sandbox is switched over to taking a list of sock_filter rather
than a list of sycall NRs, and then the generated filter programs
and linked to individual tiles and loaded at init time.

Fixes #656

### CODE REVIEW

**Review by marcus-ares — APPROVED** [review]

File: src/tango/quic/fd_quic.h [hunk 1]
```diff
@@ -478,7 +476,8 @@ fd_quic_footprint( fd_quic_limits_t const * limits );
 
 FD_QUIC_API void *
```
Comment (ptaffet-jump) [inline src/tango/quic/fd_quic.h:477]: Can you add a comment here about the ip ownership changing. Something like: `fd_quic_new` retains a read/write interest in `ip` that persists until the memory is unformatted.

File: src/tango/quic/tests/arp/test_quic_arp_server.c [hunk 2]
```diff
@@ -128,8 +128,6 @@ main( int argc, char ** argv ) {
     .stream_cnt       = { 2, 2, 2, 2 },
     .inflight_pkt_cnt = 1024,
     .tx_buf_sz        = 1<<14,
-    .routing_entries  = 32,
-    .arp_entries      = 32
```
Comment (ptaffet-jump) [inline src/tango/quic/tests/arp/test_quic_arp_server.c:132]: Is there a reason to get rid of these and only use hardcoded 256?

File: src/util/sandbox/fd_sandbox.c [hunk 3]
```diff
@@ -128,41 +128,12 @@ check_fds( ulong allow_fds_cnt,
...
-  /* none of the syscalls approved were matched: die */
-  struct sock_filter kill = BPF_STMT(BPF_RET | BPF_K, SECCOMP_RET_KILL_PROCESS);
-  filter[ 4+(allow_syscalls_cnt*2) ] = kill;
-
-  FD_TESTV( 0 == prctl( PR_SET_NO_NEW_PRIVS, 1, 0, 0, 0 ) );
-
-  struct sock_fprog default_prog = {
-    .len = (ushort)(5+(allow_syscalls_cnt*2)),
-    .filter = filter,
+install_seccomp( ushort allow_syscalls_cnt, struct sock_filter * allow_syscalls ) {
```
Comment (ptaffet-jump) [inline src/util/sandbox/fd_sandbox.c:131]: Do you want to change the names of these arguments?

File: src/util/sandbox/fd_sandbox.h [hunk 4]
```diff
@@ -67,14 +70,13 @@
 /* fd_sandbox sandboxes the current process, performing both privileged and
    private steps. */
 void
-fd_sandbox( int    full_sandbox,
```
Comment (ptaffet-jump) [inline src/util/sandbox/fd_sandbox.h:70]: Could you update the comment for this function? It has gotten a little stale.

File: src/util/sandbox/fd_sandbox.h [hunk 5]
```diff
@@ -1,10 +1,13 @@
 #ifndef HEADER_fd_src_util_sandbox_fd_sandbox_h
 #define HEADER_fd_src_util_sandbox_fd_sandbox_h
 
+#ifndef FD_HAS_FFI
```
Comment (ptaffet-jump) [inline src/util/sandbox/fd_sandbox.h:4]: Does this also need `HOSTED`?

**Review by ptaffet-jump — APPROVED** [review]
I did not review any of the seccomppolicies themselves, the header files they generate, or the filter compiler. I reviewed the rest, and it looks good. Only minor comments.

File: src/tango/quic/tests/arp/test_quic_arp_server.c (hunk 2 above)
Comment (mmcgee-jump) [reply src/tango/quic/tests/arp/test_quic_arp_server.c:132]: The caller configures them now, not QUIC. This caller could set 32 if it wanted.

**Review by ptaffet-jump — APPROVED** [review]

**Pull request merged**